cirq-core/cirq/sim/classical_simulator_test.py

```diff
@@ -78,6 +78,87 @@ def test_CCNOT():
     np.testing.assert_equal(results, expected_results)
 
 
+def test_CCCX():
```

Comment:
instead of building one big circtuit for all 8 possiblities prefer to use

```py3
@pytest.mark.parameterize(['intial_state', 'final_state'], [...])
def test_CCCX(intial_state, final_state): 
```

this way you don't need to add the any of the X layers, instead feed the initial_state to the simulator `.simulate(..., initial_state = f'{initial_state:04b}')`, that is use `.simulate` instead of `.run`

note that the initial_state should have 4 bits (1bit per qubit)

Reply:
ah, got it. I was just following the example of the other tests for ClassicalStateSimulator. Happy to make this modification though, I agree it is much cleaner! :-)

Reply:
thanks for this tip---this is much cleaner!